capstone/fabfile.py

```diff
@@ -804,3 +804,106 @@ def report_multiple_jurisdictions(out_path="court_jurisdictions.csv"):
+        "nh": { "label": "New Hampshire", "map_id": "US-NH"},
+        "pa": { "label": "Pennsylvania", "map_id": "US-PA"},
+        "mont": { "label": "Montana", "map_id": "US-MT"},
+        "ind": { "label": "Indiana", "map_id": "US-IN"},
+        "la": { "label": "Louisiana", "map_id": "US-LA"},
+        "wis": { "label": "Wisconsin", "map_id": "US-WI"},
+        "nj": { "label": "New Jersey", "map_id": "US-NJ"},
+        "ga": { "label": "Georgia", "map_id": "US-GA"},
+        "sd": { "label": "South Dakota", "map_id": "US-SD"},
+        "mass": { "label": "Massachusetts", "map_id": "US-MA"},
+        "miss": { "label": "Mississippi", "map_id": "US-MS"},
+        "cal": { "label": "California", "map_id": "US-CA"},
+        "okla": { "label": "Oklahoma", "map_id": "US-OK"},
+        "nd": { "label": "North Dakota", "map_id": "US-ND"},
+        "vt": { "label": "Vermont", "map_id": "US-VT"},
+        "ariz": { "label": "Arizona", "map_id": "US-AZ"},
+        "w-va": { "label": "West Virginia", "map_id": "US-WV"},
+        "mich": { "label": "Michigan", "map_id": "US-MI"},
+        "utah": { "label": "Utah", "map_id": "US-UT"},
+        "idaho": { "label": "Idaho", "map_id": "US-ID"},
+        "wyo": { "label": "Wyoming", "map_id": "US-WY"},
+        "colo": { "label": "Colorado", "map_id": "US-CO"},
+        "ny": { "label": "New York", "map_id": "US-NY"},
+        "ky": { "label": "Kentucky", "map_id": "US-KY"},
+        "kan": { "label": "Kansas", "map_id": "US-KS"},
+        "alaska": { "label": "Alaska", "map_id": "US-AK"},
+        "fla": { "label": "Florida", "map_id": "US-FL"},
+        "or": { "label": "Oregon", "map_id": "US-OR"},
+        "tenn": { "label": "Tennessee", "map_id": "US-TN"},
+        "md": { "label": "Maryland", "map_id": "US-MD"},
+        "ill": { "label": "Illinois", "map_id": "US-IL"},
+        "ohio": { "label": "Ohio", "map_id": "US-OH"},
+        "ala": { "label": "Alabama", "map_id": "US-AL"},
+        "sc": { "label": "South Carolina", "map_id": "US-SC"},
+        "ar": { "label": "Arkansas", "map_id": "US-AR"},
+        "ri": { "label": "Rhode Island", "map_id": "US-RI"},
+        "minn": { "label": "Minnesota", "map_id": "US-MN"},
+        "neb": { "label": "Nebraska", "map_id": "US-NE"},
+        "conn": { "label": "Connecticut", "map_id": "US-CT"},
+        "me": { "label": "Maine", "map_id": "US-ME"},
+        "iowa": { "label": "Iowa", "map_id": "US-IA"},
+        "tex": { "label": "Texas", "map_id": "US-TX"},
+        "del": { "label": "Delaware", "map_id": "US-DE"},
+        "mo": { "label": "Missouri", "map_id": "US-MO"},
+        "haw": { "label": "Hawaii", "map_id": "US-HI"},
+        "nm": { "label": "New Mexico", "map_id": "US-NM"},
+        "wash": { "label": "Washington", "map_id": "US-WA"},
+        "va": { "label": "Virginia", "map_id": "US-VA"}
+    }
+
+    tally = {}
+    for volume in VolumeMetadata.objects.prefetch_related("case_metadatas__jurisdiction", 'volume_xml__page_xmls').all():
```

Comment:
I think this loop isn't going to work for a couple of reasons. The `prefetch_related` is going to try to pull hundreds of gigabytes of case metadata and page xml into RAM, which will fail. The inner loop over cases also doesn't get the right answers -- for example 'volume_count' ends up as the number of cases, and 'case_count' ends up as the number of cases squared.

I think instead we want to do this as an aggregate sql query, or a few of them, on the capdb_jurisdiction table. Could either use a django Aggregate query (which are always frustrating) or a raw sql call. Happy to chat about this if it's helpful ...

Reply:
Ah, good point. I'll change that and set up another PR.